contracts/api3-server-v1/proxies/ProxyFactoryZkSync.sol

```diff
@@ -0,0 +1,243 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./DataFeedProxy.sol";
+import "./DapiProxy.sol";
+import "./DataFeedProxyWithOev.sol";
+import "./DapiProxyWithOev.sol";
+import "./interfaces/IProxyFactory.sol";
+
+/// @title Contract factory that deterministically deploys proxies that read
+/// data feeds (Beacons or Beacon sets) or dAPIs, along with optional OEV
+/// support
+/// @notice This ProxyFactory implementation replaces the OpenZeppelin CREATE2
+/// address computation to match zkSync's. Accordingly, only use this contract
+/// on the zkSync chains that use the ContractDeployer contract to deploy
+/// contracts. For more information, see the link below
+/// https://github.com/matter-labs/era-system-contracts/blob/857da713d9756ec13f071916848e283e74579817/contracts/ContractDeployer.sol
+/// @dev The proxies are deployed normally and not cloned to minimize the gas
+/// cost overhead while using them to read data feed values
+contract ProxyFactoryZkSync is IProxyFactory {
+    // https://github.com/matter-labs/era-system-contracts/blob/857da713d9756ec13f071916848e283e74579817/contracts/Constants.sol#L74
+    bytes32 private constant CREATE2_PREFIX =
+        0x2020dba91b30cc0006188af794c2fb30dd8520db7e2c088b7fc7c103c00ca494;
+    // According to the zkSync docs, bytecode is hashed using sha256, and then the first two bytes are
+    // replaced with the length of the bytecode in 32-byte words.
+    // To verify the hashes below
+    // - Build https://github.com/api3dao/airnode-protocol-v1/tree/deploy-zksync-reference
+    // - Get the bytecode from the artifacts
+    // - Use `utils.hashBytecode()` from the `zksync-web3` package
+    bytes32 private constant DAPI_PROXY_BYTECODE_HASH =
+        0x010000718e160c49f26d36ffd29dbe562fcc1ae0c45e3add4ae314721c4cfd50;
+    bytes32 private constant DATA_FEED_PROXY_BYTECODE_HASH =
+        0x01000071aa077a2b3722b686ce72da1b80c036fe00b90b1b0666cf7472ed7181;
+    bytes32 private constant DAPI_PROXY_WITH_OEV_BYTECODE_HASH =
+        0x010000833ea8eec6c5a363e8de8e0a9fcd770e93f86d9ec426c1f7886822cb4d;
+    bytes32 private constant DATA_FEED_PROXY_WITH_OEV_BYTECODE_HASH =
+        0x010000832145787c75d77acc93c6b6e61af2909128377978cb54e6f31e139cc0;
```

Comment:
I think it's unlikely for a potential change to be addressable by only changing these hashes. It will likely require `computeCreate2Address()` to be updated, so there will have to be a ProxyFactoryZkSyncV2 anyway.